pkg/ocm/client.go

```diff
@@ -0,0 +1,87 @@
+package ocm
+
+import (
+	"fmt"
+	"time"
+
+	sdkClient "github.com/openshift-online/ocm-sdk-go"
+	"github.com/patrickmn/go-cache"
+)
+
+type Client struct {
+	config     *Config
+	logger     sdkClient.Logger
+	connection *sdkClient.Connection
+
+	Authentication OCMAuthentication
+}
+
+type Config struct {
+	BaseURL      string `envconfig:"OCM_BASE_URL" default:""`
+	ClientID     string `envconfig:"OCM_SERVICE_CLIENT_ID" default:""`
+	ClientSecret string `envconfig:"OCM_SERVICE_CLIENT_SECRET" default:""`
+	SelfToken    string `envconfig:"OCM_SELF_TOKEN" default:""`
+	Debug        bool   `envconfig:"OCM_DEBUG" default:"false"`
+}
+
+func NewClient(config Config) (*Client, error) {
+	// Create a logger that has the debug level enabled:
+	logger, err := sdkClient.NewGoLoggerBuilder().
+		Debug(config.Debug).
+		Build()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to build OCM logger: %s", err.Error())
+	}
+	c := cache.New(1*time.Hour, 30*time.Minute)
```

Comment:
long cache? is it ok to have such?

Reply:
Yes it is fine, we were told even 12 or 24 hours is fine (but let's leave it like this)